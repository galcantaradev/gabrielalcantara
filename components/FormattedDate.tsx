import { parseISO, format } from 'date-fns';
import ptBr from 'date-fns/locale/pt-BR';

interface Props {
  value: string;
}

export const FormattedDate = (props: Props) => {
  return (
    <>
      {format(parseISO(props.value), 'dd/MM/yyyy', {
        locale: ptBr
      })}
    </>
  );
};
