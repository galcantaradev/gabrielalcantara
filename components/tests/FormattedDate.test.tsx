import { render, screen } from '@testing-library/react';

import { FormattedDate } from '../FormattedDate';

describe('FormattedDate.test.tsx', () => {
  it('should render formatted date', () => {
    const { asFragment } = render(<FormattedDate value="2021-01-01" />);

    expect(screen.getByText('01/01/2021')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
