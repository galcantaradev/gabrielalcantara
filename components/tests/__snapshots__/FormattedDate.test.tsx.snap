// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedDate.test.tsx should render formatted date 1`] = `
<DocumentFragment>
  01/01/2021
</DocumentFragment>
`;
